File: CSharpPractice/LeetcodePractice/Leetcode45.cs
public class Solution45 {
    public int Jump(int[] nums) 
    {
        int step = 0;
        int curRight = 0;
        int maxRight = 0;
        for (int i = 0; i < nums.Length; i++)
        {
            if (i>curRight)
            {
                step++;
                curRight = maxRight;
            }
            if (maxRight<nums[i]+i)
            {
                maxRight = nums[i]+i;
                if (maxRight >= nums.Length-1 && i<nums.Length-1)
                {
                    return step;
                }
            }
        }
        return step+1;
        
    }
    
}

File: JavaScriptPractice/Leetcode45.js
var jump = function(nums) 
{
    let curRight = 0;
    let maxRight = 0;
    let step = 0;
    for (let i = 0; i < nums.length; i++) {
        if (i>curRight) 
        {
            curRight = maxRight;
            step++;
        }
        if (maxRight<nums[i]+i) 
        {
            maxRight=nums[i]+i;
            if (maxRight>=nums.length-1 && i!=nums.length-1) 
            {
                return step+1;
            }
        }
    }
    return step;
};